docs/guide/philosophy.md

```diff
@@ -0,0 +1,27 @@
+# Project Philosophy
+
+## Lean Extendable Core
+
+Vite doesn't intend to cover every use case for every user. Vite aims to support the most common patterns to build Web apps out-of-the-box, but [Vite core](https://github.com/vitejs/vite) must remain lean with a small API surface to keep the project maintainable long-term. This goal is possible thanks to [Vite's rollup-based plugin system](./api-plugin.md). Features that can be implemented as external plugins will generally not be added to Vite core. [vite-plugin-pwa](https://vite-pwa-org.netlify.app/) is a great example of what can be achieved out of Vite core, and there are a lot of [well maintained plugins](https://github.com/vitejs/awesome-vite#plugins) to cover your needs. Vite works closely with the Rollup project to ensure that plugins can be used in both plain-rollup and Vite projects as much as possible, trying to push needed extensions to the Plugin API upstream when possible.
+
+## Pushing the Modern Web
+
+Vite provides opinionated features that pushes writing modern code. For example:
+
+- The source code can only be written in ESM, where non-ESM dependencies needs to be [pre-bundled as ESM](./dep-pre-bundling) in order to work.
+- Web workers are encouraged to be written with the [`new Worker` syntax](./features#web-workers) to follow modern standards.
+- Node.js modules cannot be used in the browser.
+
+When adding new features, these patterns are followed to create a future-proof API, rather than following the competition.
```

Comment:
Yeah rewording that sounds fine to me. Maybe it can be shorter like:
```suggestion
When adding new features, these patterns are followed to create a future-proof API, which may not always be compatible with other build tools.
```